### DISCUSSION TITLE
[Question] Why is the dom::tape limited to 64-bit integers?

### DISCUSSION THREAD

**Discussion by a-soll** [simdjson/simdjson#2250] [Q&A]
I was messing with it briefly and it seems like it might be possible to store 32-bit ints in a `tape_ref`'s `string_buf` index. It also seems as though it wouldn't affect current 64-bit usage because it would only utilize 32-bit ints if the caller is trying to cast to a 32-bit type. And in majority of cases, the int being read out of the json is going to be small enough to fit in just an `int32_t`. Are there benefits to being purely 64-bit?

**Comment by lemire** [comment]
@a-soll 


You should implement your alternative design and demonstrate its benefits.

The library is community driven. If you contribute a pull request and you document the clear benefits of your approach, it will be given full consideration. 



Do you expect significant performance benefits?
